src/main/java/life/qbic/data/processing/scanner/Scanner.java

```diff
@@ -102,15 +102,28 @@ public void run() {
   private List<RegistrationRequest> detectDataForRegistration() {
     return userProcessDirectories.parallelStream()
         .map(Path::toFile)
+        .filter(this::matchesAccessRightsCriteria)
         .map(file -> createRequests(file.listFiles(), file.toPath())).flatMap(
             Collection::stream).toList();
   }
 
+  private boolean matchesAccessRightsCriteria(File file) {
+    if (!file.canWrite()) {
+      log.error("Cannot write to file '{}'", file);
+      return false;
+    }
+    if (!file.canExecute()) {
+      log.error("Cannot execute file '{}'", file);
+      return false;
+    }
+    return true;
+  }
+
   private List<RegistrationRequest> createRequests(File[] files, Path userDirectory) {
     if (files == null || files.length == 0) {
       return new ArrayList<>();
     }
-    return Arrays.stream(files).filter(file -> !file.isHidden())
+    return Arrays.stream(files).filter(file -> !file.isHidden()).filter(File::isDirectory)
```

Comment:
I would argue that the filtering for directories should take place earlier. In the `detectDataForRegistration` there is a variable `userProcessDirectories` containing both files that are and file that are not directories?
Why is the check in line 74 not sufficient?

Reply:
Good point!

I added another filter method and apply all filters before the `createRequest`  method, since it violates the definition of a function.